tests/timelock-controller.tests.ts

```diff
@@ -0,0 +1,172 @@
+import {bnArg, expect, fromSigner, setupContract} from './helpers'
+import {network, patract} from 'redspot'
+import BN from "bn.js";
+
+const {getRandomSigner} = patract
+const {api, getSigners} = network
+
+describe('MY_TIMELOCK_CONTROLLER', () => {
+    async function setup() {
+        const one = new BN(10).pow(new BN(api.registry.chainDecimals[0]))
+        const signers = await getSigners()
+        const bob = await getRandomSigner(signers[1], one.muln(10000))
+        let contract = await setupContract('my_timelock_controller', 'new', 0, [bob.address], [bob.address]);
+
+        return { contract, bob }
+    }
+
+    it('TIMELOCK CONTROLLER - can schedule', async () => {
+        const {
+            contract,
+            bob,
+        } = await setup()
+
+        // Arrange - Prepare data for schedule
+        let transaction = {
+            callee: contract.contract.address,
+            selector: [0, 0, 0, 0],
+            input: [],
+            transferred_value: 0,
+            gas_limit: 0,
+        }
+        let salt = bnArg(0);
+
+
+        // Act - Bob scheduled the transaction
+        // @ts-ignore
+        let id = (await contract.query.hashOperation(transaction, null, salt)).output!;
+        await expect(contract.query.isOperationPending(id)).to.have.output(false)
+        // @ts-ignore
+        await expect(fromSigner(contract.contract, bob.address).tx.schedule(transaction, null, salt, 0)).to.eventually.be.fulfilled
+
+        // Assert - Operation must be scheduled, it should be in Pending state and in Ready state(because min delay is zero)
+        await expect(contract.query.isOperationPending(id)).to.have.output(true)
+        await expect(contract.query.isOperationReady(id)).to.have.output(true)
+        await expect(contract.query.isOperationDone(id)).to.have.output(false)
+    })
+
+    it('TIMELOCK CONTROLLER - schedule and execute without input data `TimelockController,get_min_delay`', async () => {
+        const {
+            contract,
+            bob,
+        } = await setup()
+
+        // Arrange - Prepare data for execute `get_min_delay`
+        const message = contract.contract.abi.messages.find(message => message.identifier === "TimelockController,get_min_delay")!;
```

Comment:
can we add helper method: const {selector} =  getMessageAbi(contract.contract, "TimelockController,get_min_delay")